addons/gamification/controllers/main.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import werkzeug.exceptions
+import werkzeug.urls
+import werkzeug.wrappers
+
+from odoo import http, modules
+from odoo.http import request
+
+
+class GamificationController(http.Controller):
+
+    def _get_image(self, model, record_id):
+        status, headers, content = request.env['ir.http'].sudo().binary_content(
+            model=model, id=record_id, field='image', default_mimetype='image/png')
+
+        if not content:
+            img_path = modules.get_module_resource('web', 'static/src/img', 'placeholder.png')
+            with open(img_path, 'rb') as f:
+                image = f.read()
+            content = base64.b64encode(image)
+        if status == 304:
+            return werkzeug.wrappers.Response(status=304)
+        image_base64 = base64.b64decode(content)
+        headers.append(('Content-Length', len(image_base64)))
+        response = request.make_response(image_base64, headers)
+        response.status = str(status)
+        return response
+
+    @http.route(['/gamification/badge/<model("gamification.badge"):badge>/icon'], type='http', auth="public", website=True, sitemap=False)
+    def get_badge_icon(self, badge, **post):
```

Comment:
Iwould remove all those routes and methods and simply use /web/image . Normally everything is already supported :) . A lot of templates use them :) .